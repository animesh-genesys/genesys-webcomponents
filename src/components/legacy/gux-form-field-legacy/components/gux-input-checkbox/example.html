<form onchange="notify(event)" oninput="notify(event)">
  <h1>gux-form-field-legacy (Checkbox)</h1>

  <p>
    slot a input[type=checkbox] element into a gux-form-field-legacy element
  </p>

  <div>
    <gux-form-field-legacy>
      <input slot="input" type="checkbox" name="food" value="pizza" />
      <label slot="label">Pizza</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="checkbox" name="food" value="pasta" checked />
      <label slot="label">Pasta</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input
        slot="input"
        type="checkbox"
        name="food"
        value="hamburger"
        id="hamburger"
      />
      <label slot="label">Hamburger</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input
        slot="input"
        type="checkbox"
        name="food"
        value="sandwich"
        disabled
      />
      <label slot="label">Sandwich</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="checkbox" name="food" value="sushi" />
      <label slot="label">Sushi</label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>
</form>

<style
  onload="(function () {
  document.getElementById('hamburger').indeterminate = true;
})()"
>
  .not-used {
    -custom-noop: noop;
  }
</style>
