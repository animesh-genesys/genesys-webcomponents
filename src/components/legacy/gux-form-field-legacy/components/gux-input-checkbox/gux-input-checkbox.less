@import (reference) '../../../../../style/color.less';

@gux-icon-checkbox-unchecked: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 2.467a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7m0-1h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3z' fill='%2377828f' /%3E%3C/svg%3E");
@gux-icon-checkbox-unchecked-hover: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 2.467a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7m0-1h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3z' fill='%232a60c8' /%3E%3C/svg%3E");
@gux-icon-checkbox-unchecked-error: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 2.467a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7m0-1h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3z' fill='%23ea0b0b' /%3E%3C/svg%3E");
@gux-icon-checkbox-checked: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 1.467h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3zM6.8 11 3.638 7.941l.89-.932L6.8 9.2l3.947-3.861.891.932z' fill='%232a60c8' /%3E%3C/svg%3E");
@gux-icon-checkbox-mixed: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M4.138 8.467h7v-1h-7zm7-7h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3zm2 10a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7a2 2 0 0 1 2 2z' fill='%232a60c8' /%3E%3C/svg%3E");

gux-input-checkbox {
  display: block;
  color: @gux-black-50;

  .gux-input-checkbox-container {
    position: relative;
    padding-left: 24px;
    line-height: 24px;
  }

  input {
    position: absolute;
    z-index: -1;
    opacity: 0;
  }

  label {
    display: inline-block;
    font-size: 12px;

    &:hover {
      cursor: pointer;
    }
  }

  label::after {
    position: absolute;
    top: 4px;
    left: 4px;
    display: block;
    width: @gux-icon-size-default-global;
    height: @gux-icon-size-default-global;
    content: '';
    border-radius: 15%;
  }

  input:focus-within ~ label::after {
    box-shadow: 0 0 3px 2px @gux-blue-90;
  }

  input:not(:checked) ~ label::after {
    background-image: @gux-icon-checkbox-unchecked;
  }

  &.gux-input-error input:not(:checked) ~ label::after {
    background-image: @gux-icon-checkbox-unchecked-error;
  }

  input:not(:checked):not(:disabled):not(:indeterminate) ~ label:hover::after {
    background-image: @gux-icon-checkbox-unchecked-hover;
  }

  input:checked ~ label::after {
    background-image: @gux-icon-checkbox-checked;
  }

  input:indeterminate ~ label::after {
    background-image: @gux-icon-checkbox-mixed;
  }

  input:disabled ~ label::after,
  input:disabled ~ label {
    cursor: not-allowed;
    opacity: 0.5;
  }
}
