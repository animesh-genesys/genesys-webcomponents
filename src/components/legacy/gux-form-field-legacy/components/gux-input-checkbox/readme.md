# gux-input-checkbox

<!-- Auto Generated Below -->


## Slots

| Slot      | Description                              |
| --------- | ---------------------------------------- |
| `"input"` | Required slot for input[type="checkbox"] |
| `"label"` | Required slot for label                  |


## Dependencies

### Used by

 - [gux-form-field-legacy](../..)

### Graph
```mermaid
graph TD;
  gux-form-field-legacy --> gux-input-checkbox
  style gux-input-checkbox fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
