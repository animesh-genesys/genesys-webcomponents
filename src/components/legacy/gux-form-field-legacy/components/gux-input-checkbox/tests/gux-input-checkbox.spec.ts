import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { GuxInputCheckbox } from '../gux-input-checkbox';

describe('gux-input-checkbox', () => {
  let page: SpecPage;

  beforeEach(async () => {
    page = await newSpecPage({
      components: [GuxInputCheckbox],
      html: `
        <gux-input-checkbox>
          <input slot="input" type="checkbox" id="pizza" name="food" value="pizza">
          <label slot="label" for="pizza">Pizza</label>
        </gux-input-checkbox>
      `,
      language: 'en'
    });
  });

  it('should build', async () => {
    expect(page.rootInstance).toBeInstanceOf(GuxInputCheckbox);
  });

  it('should render', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
