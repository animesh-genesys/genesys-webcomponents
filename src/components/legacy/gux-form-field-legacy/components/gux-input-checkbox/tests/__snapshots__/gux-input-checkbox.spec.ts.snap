// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-input-checkbox should render 1`] = `
<gux-input-checkbox>
  <!---->
  <div class="gux-input-checkbox-container">
    <input id="pizza" name="food" slot="input" type="checkbox" value="pizza">
    <label for="pizza" slot="label">
      Pizza
    </label>
  </div>
</gux-input-checkbox>
`;
